crates/bin/c8y-device-management/src/main.rs

```diff
@@ -68,6 +74,11 @@ async fn main() -> anyhow::Result<()> {
     log_actor.with_c8y_http_proxy(&mut c8y_http_proxy_actor)?;
     log_actor.with_mqtt_connection(&mut mqtt_actor)?;
 
+    //Instantiate health monitor actor
+    let mut health_actor = HealthMonitorBuilder::new(PLUGIN_NAME);
+    health_actor.set_connection(&mut mqtt_actor);
+    // health_actor.with_mqtt_connection(&mut mqtt_actor)?;
```

Comment:
This is correct. However `mut` can be avoided using the `with_connection` method. 


```suggestion
    let health_actor = HealthMonitorBuilder::new(PLUGIN_NAME)
             .health_actor.with_connection(&mut mqtt_actor);
```